tools/check_stamped.py

```diff
@@ -0,0 +1,148 @@
+#  all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+#  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+#  THE SOFTWARE.
+#####################################################################################
+import subprocess
+import sys
+
+debug = False
+
+def hasKeySequence(inputfile, key_message):
+    result = False
+    if key_message in inputfile:
+        result = True
+
+    return result
+
+
+#Simple just open and write stuff to each file with the license stamp
+def openAndCheckFile(filename):
+    result = False
+    #open save old contents and append things here
+    if debug is True:
+        print("Open", filename, end='')
+
+    try:
+        file = open(filename, 'r')
+    except OSError as e:
+        if debug is True:
+            print(str(e) + "....Open Error: Skipping  file ")
+        file.close()
+        return
+    else:
+        with file as contents:
+            try:
+                save = contents.read()
+                hasAmdLic = hasKeySequence(
+                    save, "Advanced Micro Devices, Inc. All rights reserved")
+
+                #Check if we have a licence stamp already
+                if hasAmdLic is True:
+                    if debug is True:
+                        print("....Already Stamped: Skipping  file ")
+                    contents.close()
+                    result = True
+
+            except UnicodeDecodeError as eu:
+                if debug is True:
+                    print(str(eu) + "...Skipping binary file ")
+                contents.close()
+                result = True
+
+    return result
+
+
+# Deterine if filename is desired in the fileTuple past in
+def check_filename(filename, fileTuple):
+    supported = False
+
+    for key in fileTuple:
+        if key in filename:
+            supported = True
+            break
+
+    return supported
+
+def main():
+    # The filetypes we want to check for that are stamped
+    # LICENSE is included here as it SHOULD have a liscence in it otherwise flag it as unstamped
+    supported_file_types = (".cpp", ".hpp", ".h", ".ipynb", ".py", ".txt",
+                            ".sh", ".bsh", "LICENSE")
+
+    #add general stuff we shouldn't stamp and any exceptions here
+    unsupported_file_types = [
+        ".onnx", ".pb", ".rst", ".jpg", ".jpeg", ".proto", ".md", ".clang",
+        ".cmake", ".weight", ".ini", ".json", ".docker", ".git", ".rules",
+        ".yml"
+    ]
+
+    #Specifics to ignore
+    specificIgnores = ("digits.txt", "Dockerfile", "Jenkinsfile", "")
```

Comment:
Maybe we should move these variables up to the top, since this is probably what we would change in the future.

Reply:
all three of them.